onedal/decomposition/tests/test_incremental_pca.py

```diff
@@ -0,0 +1,88 @@
+# ==============================================================================
+# Copyright 2024 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+import numpy as np
+import pytest
+from numpy.testing import assert_allclose
+
+from daal4py.sklearn._utils import daal_check_version
+from onedal.decomposition import IncrementalPCA
+from onedal.tests.utils._device_selection import get_queues
+
+
+@pytest.mark.parametrize("queue", get_queues())
+@pytest.mark.parametrize("is_deterministic", [True, False])
+@pytest.mark.parametrize("num_blocks", [1, 2, 3])
+@pytest.mark.parametrize("dtype", [np.float32, np.float64])
+def test_on_gold_data(queue, is_deterministic, num_blocks, dtype):
+    X = np.array([[-1, -1], [-2, -1], [-3, -2], [1, 1], [2, 1], [3, 2]])
+    X = X.astype(dtype=dtype)
+    X_split = np.array_split(X, num_blocks)
+    incpca = IncrementalPCA()
+
+    for i in range(num_blocks):
+        incpca.partial_fit(X_split[i], queue=queue)
+
+    result = incpca.finalize_fit()
+
+    transformed_data = incpca.predict(X, queue=queue)
+
+    expected_n_components_ = 2
+    expected_components_ = np.array([[0.83849224, 0.54491354], [-0.54491354, 0.83849224]])
+    expected_singular_values_ = np.array([6.30061232, 0.54980396])
+    expected_mean_ = np.array([0, 0])
+    expected_explained_variance_ = np.array([7.93954312, 0.06045688])
+    expected_explained_variance_ratio_ = np.array([0.99244289, 0.00755711])
+    expected_transformed_data = np.array(
```

Comment:
I highly recommend to compute it on flight using just NumPy functionality

Reply:
I agree